src/libpriv/rpmostree-rpm-util.c

```diff
@@ -836,6 +836,43 @@ rpmostree_get_refsack_for_root (int              dfd,
   return rpmostree_refsack_new (sack, NULL);
 }
 
+/* Given @dfd + @path, return a sack corresponding to the base layer (which is the same as
+ * /usr/share/rpm if it's not a layered deployment.
+ *
+ * Note this function may return %TRUE without a sack if the deployment predates when we
+ * started embedding RPMOSTREE_BASE_RPMDB.
+ */
+gboolean
+rpmostree_get_base_refsack_for_root (int                dfd,
+                                     const char        *path,
+                                     RpmOstreeRefSack **out_sack,
+                                     GCancellable      *cancellable,
+                                     GError           **error)
+{
+  g_autofree char *subpath = g_build_filename (path, RPMOSTREE_BASE_RPMDB, NULL);
+  if (!glnx_fstatat_allow_noent (dfd, subpath, NULL, AT_SYMLINK_NOFOLLOW, error))
+    return FALSE;
+  if (errno == ENOENT)
+    return TRUE;
+
+  g_auto(GLnxTmpDir) tmpdir = {0, };
+  if (!glnx_mkdtemp ("rpmostree-dbquery-XXXXXX", 0700, &tmpdir, error))
+    return FALSE;
+  if (!glnx_shutil_mkdir_p_at (tmpdir.fd, "var/lib", 0777, cancellable, error))
+    return FALSE;
+
+  g_autofree char *base_rpm = glnx_fdrel_abspath (dfd, subpath);
+  if (symlinkat (base_rpm, tmpdir.fd, "var/lib/rpm") == -1)
```

Comment:
This is pretty ugly.  I'm sure we have similar code elsewhere, just a reminder to ourselves to eventually teach librpm to find the rpmdb in the place we want it to start with.